### ISSUE TITLE
Using TableSetupScrollFreeze causes MultiSelect to skip items when scrolling upwards in a Table

### ISSUE THREAD

**Issue by bratpilz** [ocornut/imgui#7821]
### Version/Branch of Dear ImGui:

Version 1.91.0 WIP (19099), Branch: docking

### Back-ends:

imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp

### Compiler, OS:

Linux + GCC

### Full config/build information:

```
Dear ImGui 1.91.0 WIP (19099)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=201103
define: __linux__
define: __GNUC__=12
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: imgui_impl_glfw
io.BackendRendererName: imgui_impl_opengl3
io.ConfigFlags: 0x00000483
 NavEnableKeyboard
 NavEnableGamepad
 DockingEnable
 ViewportsEnable
io.ConfigViewportsNoDecoration
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x00001C0E
 HasMouseCursors
 HasSetMousePos
 PlatformHasViewports
 HasMouseHoveredViewport
 RendererHasVtxOffset
 RendererHasViewports
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1280.00,720.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

### Details:

**My Issue/Question:**

Hello! While trying out the new MultiSelect functionality, I noticed that it doesn't play nice with TableSetupScrollFreeze as of now, and the box selection skips items when scrolling upwards. You can see what I mean in the attached example video and code.
Can you see an easy way to make this work? Thanks in advance!

### Screenshots/Video:

https://github.com/user-attachments/assets/4cbbfbee-f06f-458f-beb4-e8b2dfc6447c

### Minimal, Complete and Verifiable Example code:

```cpp
// Here's some code anyone can copy and paste to reproduce your issue
	{
	  using namespace ImGui;
	  SetNextWindowSize (ImVec2 (400.0f, 300.0f), ImGuiCond_Appearing);
	  Begin ("Test");
	  static bool freeze_header = true;
	  Checkbox ("Freeze header", &freeze_header);
	  static ImGuiSelectionBasicStorage selection;
	  if (BeginTable ("Table", 1, ImGuiTableFlags_ScrollY, GetContentRegionAvail()))
	    {
	      TableSetupColumn ("Column");
	      if (freeze_header)
		TableSetupScrollFreeze (0, 1);
	      TableHeadersRow();

	      static unsigned const N = 100;
	      auto ms = BeginMultiSelect (ImGuiMultiSelectFlags_BoxSelect1d, selection.Size, N);
	      selection.ApplyRequests (ms);

	      for (unsigned i = 0; i < N; ++i)
		{
		  TableNextColumn();
		  SetNextItemSelectionUserData (i);
		  char label[32];
		  snprintf (label, sizeof label, "Row %d", i);
		  Selectable (label, selection.Contains (i));
		}

	      ms = EndMultiSelect();
	      selection.ApplyRequests (ms);

	      EndTable();
	    }
	  End();
	}
```

**Comment by ocornut** [comment]
It seems like the edge of the box-select rectangle should be corrected to exclude the frozen rows and be limited to the scrollable region.

It's important that this rectangle is limited to the correct area as box-select does a selection/unselect when an item cross the boundaries between old and new rectangles.

**Comment by ocornut** [comment]
This is analogous to the fix we did for #5143 and namely 0b4a1a401, for some reason adding `+ ImVec2(window->DecoInnerSizeX1, window->DecoInnerSizeY1)` to scope_rect in `EndMultiSelect()` doesn't seem to be enough, I'll try to look at this on Monday when I have more time.

**Comment by ocornut** [comment]
Should be fixed by 249d5ca and b847c41. Thanks for reporting!
